source/frontend/src/features/mapSideBar/acquisition/tabs/agreement/update/UpdateAcquisitionAgreementContainer.tsx

```diff
@@ -0,0 +1,89 @@
+import { AxiosError } from 'axios';
+import { useCallback, useEffect, useState } from 'react';
+import { useHistory, useLocation } from 'react-router-dom';
+import { toast } from 'react-toastify';
+
+import { useAgreementProvider } from '@/hooks/repositories/useAgreementProvider';
+import { useModalContext } from '@/hooks/useModalContext';
+import { IApiError } from '@/interfaces/IApiError';
+import { ApiGen_Concepts_Agreement } from '@/models/api/generated/ApiGen_Concepts_Agreement';
+
+import { IUpdateAcquisitionAgreementViewProps } from '../common/UpdateAcquisitionAgreementView';
+import { AcquisitionAgreementFormModel } from '../models/AcquisitionAgreementFormModel';
+
+export interface IUpdateAcquisitionAgreementContainerProps {
+  acquisitionFileId: number;
+  agreementId: number;
+  View: React.FC<IUpdateAcquisitionAgreementViewProps>;
+  onSuccess: () => void;
+}
+
+const UpdateAcquisitionAgreementContainer: React.FunctionComponent<
+  React.PropsWithChildren<IUpdateAcquisitionAgreementContainerProps>
+> = ({ acquisitionFileId, agreementId, View, onSuccess }) => {
+  const history = useHistory();
+  const location = useLocation();
+  const { setModalContent, setDisplayModal } = useModalContext();
+  const [initialValues, setInitialValues] = useState<AcquisitionAgreementFormModel | null>(null);
+
+  const backUrl = location.pathname.split(`/${agreementId}/update`)[0];
+
+  const {
+    updateAcquisitionAgreement: { execute: updateAcquisitionAgreement, loading: updatingAgreement },
+    getAcquisitionAgreementById: { execute: getAgreement, loading: fetchingAgreement },
+  } = useAgreementProvider();
+
+  const handleSave = async (updatedAgreement: ApiGen_Concepts_Agreement) => {
+    return updateAcquisitionAgreement(acquisitionFileId, agreementId, updatedAgreement);
+  };
+
+  const handleSuccess = async () => {
+    onSuccess();
+    history.push(backUrl);
+  };
+
+  const onCreateError = (e: AxiosError<IApiError>) => {
+    if (e?.response?.status === 409) {
+      setModalContent({
+        variant: 'error',
+        title: 'Error',
+        message: 'You can only have one offer with status of "Accepted".',
+        okButtonText: 'Close',
+        handleOk: () => setDisplayModal(false),
+      });
+      setDisplayModal(true);
```

Comment:
IMO the modal should happen on the view instead of the container

Reply:
I think it's ok to be on the container since it would need the Axios context to handle issues.